parity-clib/src/lib.rs

```diff
@@ -136,31 +159,48 @@ pub unsafe extern fn parity_rpc(client: *mut c_void, query: *const c_char, len:
 			}
 		};
 
-		if let Some(output) = client.rpc_query_sync(query_str) {
-			let q_out_len = output.as_bytes().len();
-			if *out_len < q_out_len {
-				return 1;
-			}
+		let callback = match callback {
+			Some(callback) => Arc::new(callback),
+			None => return 1,
+		};
 
-			ptr::copy_nonoverlapping(output.as_bytes().as_ptr(), out_str as *mut u8, q_out_len);
-			*out_len = q_out_len;
-			0
-		} else {
-			1
-		}
+		let cb = callback.clone();
+
+		// FIXME: provide session object here, if we want to support the `PubSub`
```

Comment:
Well, the whole point of #9271 is to implement this. It's also the most difficult part, and why I gave up doing this myself.

Reply:
> everytime we get a notification we should send it to the callback?

Yes!